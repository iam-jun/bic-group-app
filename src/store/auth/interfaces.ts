import {CognitoUser, ISignUpResult} from 'amazon-cognito-identity-js';
// Actiontype
export interface ISignIn {
  email: string;
  password: string;
}

export interface ISignUp {
  username: string;
  email: string;
  password: string;
}

export interface IForgotPassword {
  email: string;
  callback: () => void;
}

export interface IForgotPasswordRequest {
  code: string;
  email: string;
  password: string;
}

export interface IUser extends CognitoUser {}

export interface ISignUpResponse extends ISignUpResult {}
