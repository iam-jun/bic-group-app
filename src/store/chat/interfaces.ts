import {IMessage as _GMessage} from 'react-native-gifted-chat';
import {IUser} from '../auth/interfaces';
export interface IReaction {
  type: string;
  count: number;
  reacted?: boolean;
}

export type GMessage = _GMessage;

export type IAttachment = {
  name: string;
  url: string;
  thumnailUrl?: string;
  type: string;
};

export type IMessage = _GMessage & {
  id: string | number;
  user: IUser;
  quoted_message?: IMessage;
  reactions?: IReaction[];
  attachments?: IAttachment[];
};

export type IConversation = {
  name: string;
  description?: string;
  members: IUser[];
  unreadCount: number;
  lastMessage: string;
  updatedAt: string;
};
