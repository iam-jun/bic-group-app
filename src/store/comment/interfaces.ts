import {IUser} from '../auth/interfaces';

export type IComment = {
  id: string;
  content: string;
  createdAt: string;
  updateAt: string;
  user: IUser;
  replyCount: number;
};
