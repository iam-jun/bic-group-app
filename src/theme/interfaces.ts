export interface ITheme extends ReactNativePaper.Theme {
  dark: boolean;
  colors: IThemeColors;
  spacing?: ISpacing;
  dimension?: any;
  shadow?: any;
  fontFamily?: any;
  fonts: any;
  animation: any;
}

export interface IThemeColors extends ReactNativePaper.ThemeColors {
  primary: string;
  primary1?: string;
  primary2?: string;
  primary3?: string;
  primary4?: string;
  primary5?: string;
  primary6?: string;
  primary7?: string;

  background: string;
  bgFocus?: string;
  bgHover?: string;
  bgDisable?: string;
  bgButtonPrimary?: string;
  bgButtonSecondary?: string;
  border?: string;
  placeholder: string;
  surface: string;
  iconTint?: string;
  iconTintReversed?: string;

  link?: string;
  activeLink?: string;
  success?: string;
  error: string;
  warning?: string;

  textPrimary?: string;
  textSecondary?: string;
  textSuccess?: string;
  textDanger?: string;
  textWarning?: string;
  textInfo?: string;
  textDisabled?: string;
  textReversed?: string;

  facebook?: string;
  google?: string;
  apple?: string;
}

export interface ISpacing {
  margin: ISpacingValue;
  padding: ISpacingValue;
  borderRadius: ISpacingValue;
  lineHeight: ISpacingValue;
}

export interface ISpacingValue {
  base?: number;
  tiny?: number;
  small?: number;
  large?: number;
  big?: number;
}
