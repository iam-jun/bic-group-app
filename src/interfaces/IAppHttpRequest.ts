export interface IParamsGetUsers {
  key?: string;
  offset?: number;
  limit?: number;
  sort?: 'id:asc' | 'id:desc' | 'id:asc,createdAt:desc' | string;
  userIds?: string;
}
