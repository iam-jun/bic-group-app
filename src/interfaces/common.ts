import {languages} from '~/configs';
import {IconType} from '~/resources/icons';
import {StreamClient} from 'getstream';
import {ToastMessageProps} from '~/beinComponents/ToastMessage/NormalToastMessage';

export interface IObject<T> {
  [key: string]: T;
}

export interface ILanguage {
  code: keyof typeof languages;
  title: string;
  icon: IconType;
}

export interface ISetting {
  type: string;
  title: string;
  icon: IconType;
}

export interface IGetStreamDispatch {
  streamClient: StreamClient;
  userId: string;
}

export interface IRightMenu {
  type: string;
  title: string;
}

export interface IFilePicked {
  name: string;
  filename: string;
  type: string;
  mime: string;
  size: number;
  width: number;
  height: number;
  //app
  uri: string;
  //web
  base64?: string;
  [x: string]: any;
}

export interface IResponseData {
  code: number;
  data: any;
  meta: any;
}

export interface IToastMessage {
  content: string;
  props: ToastMessageProps;
  duration?: number;
}

export interface IPayloadShowModal {
  isOpen: boolean;
  ContentComponent: any;
}
