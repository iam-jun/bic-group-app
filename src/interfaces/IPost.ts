import {IFilePicked} from '~/interfaces/common';
import {ReactionType} from '~/constants/reactions';

export interface IPostAudience {
  users?: IAudienceUser[];
  groups?: IAudienceGroup[];
}

export interface IAudienceUser {
  id?: number | string;
  username?: string;
  fullname?: string;
  avatar?: string;
}

export interface IAudienceGroup {
  id?: number | string;
  name?: string;
  icon?: string;
  child?: number[];
}

export interface IMarkdownAudience {
  id: string;
  data: {
    fullname: string;
    username: string;
    avatar?: string;
  };
  created_at: string;
  updated_at: string;
}

export interface IAudience {
  id?: number | string;
  name?: string;
  icon?: string;
  avatar?: string;
  userCount?: number;
  type?: string;
  isGroup?: boolean;
}

export interface IActivityDataImage {
  name: string;
  origin_name?: string;
  width?: number;
  height?: number;
}

export interface IActivityData {
  content?: string;
  images?: IActivityDataImage[];
  videos?: string[];
  files?: string[];
  edited?: boolean;
}

export interface IActivityImportant {
  active?: boolean;
  expires_time?: string;
}

export interface ICommentData {
  id?: number;
  postId?: number;
  totalReply?: number;
  actor?: IAudienceUser;
  parentId?: number;
  content?: string;
  media?: IPostMedia;
  mentions?: any;
  createdAt?: string;
  updatedAt?: string;
  ownerReactions?: any;
  reactionsCount?: any;
  child?: any;
}

export interface ICreatePostImage {
  fileName?: string;
  file?: IFilePicked;
  url?: string;
}

export interface ICreatePostSettings {
  important?: IActivityImportant;
  count: number;
}

export interface IPostSetting {
  canReact?: boolean;
  canShare?: boolean;
  canComment?: boolean;
  isImportant?: boolean;
  importantExpiredAt?: string;
}

export interface IPostMedia {
  images?: any[];
  videos?: any[];
  files?: any[];
}

export interface IPostActivity {
  id?: string;
  audience?: IPostAudience;
  content?: string;
  highlight?: string;
  media?: IPostMedia;
  setting?: IPostSetting;
  isDraft?: boolean;
  actor?: IAudienceUser;
  mentions?: any;
  commentsCount?: number;
  reactionsCount?: IReactionCounts;
  ownerReactions?: IOwnReaction;
  createdAt?: string;
  createdBy?: number;
}

export interface IOwnReaction {
  [x: string]: {[reactionKind: string]: IReaction};
}

export type IReactionCounts = {
  [x: string]: {[reactionKind: string]: number};
};

export interface IAllPosts {
  [id: string]: IPostActivity;
}

export interface IAllComments {
  [id: string]: IReaction;
}

export interface IPostCreatePost {
  audience?: {
    userIds: number[];
    groupIds: number[];
  };
  content?: string;
  media?: any;
  setting?: any;
  mentions?: any;
  isDraft?: boolean;

  createFromGroupId?: string | number;
}

export interface IPayloadCreatePost {
  data?: IPostCreatePost;
  createFromGroupId?: string | number;
  callback?: any;
}

export interface IPayloadCreateComment {
  postId: string;
  parentCommentId?: string | number;
  commentData: ICommentData;
  userId: string | number;
  localId?: string | number[]; // used when retry adding new comment
  preComment?: IReaction & {
    localId: string | number[]; // used when creating new comment
    parentCommentId?: string | number;
  };
  onSuccess?: () => void;
  isCommentLevel1Screen?: boolean;
}

export interface IPayloadPutEditPost {
  id: string;
  data: IPostCreatePost;
  replaceWithDetail?: boolean;
  onRetry?: () => void;
}

export interface IPayloadPutEditComment {
  id: string;
  comment: IReaction;
  data: IActivityData;
}

export interface IPayloadDeletePost {
  id: string;
  isDraftPost?: boolean;
}

export interface IParamGetPostDetail {
  postId: string;
  commentOrder?: 'ASC' | 'DESC';
  commentLimit?: number;
  childCommentLimit?: number;
}

export interface IParamPutEditPost {
  postId: string;
  data: IPostCreatePost;
}

export interface IPayloadGetPostDetail extends IParamGetPostDetail {
  callbackLoading?: (loading: boolean, success: boolean) => void;
}

export type IReactionKind = 'comment' | 'seen' | ReactionType;

export interface IGetStreamUser {
  created_at?: string;
  updated_at?: string;
  id?: string | number;
  data?: {
    avatar?: string;
    fullname?: string;
    username?: string;
  };
}

export interface IRequestPostComment {
  postId: string;
  data: ICommentData;
}

export interface IRequestReplyComment {
  parentCommentId: string | number;
  data: IActivityData;
}

export interface IRequestGetPostComment {
  postId: string;
  commentId?: string;
  idLt?: string; //get comment before this id
  kind?: string;
  recentReactionsLimit?: number;
  recentChildReactionsLimit?: number;
}

export interface IPayloadGetCommentsById extends IRequestGetPostComment {
  isMerge: boolean;
  callbackLoading?: (loading: boolean, canLoadMore?: boolean) => void;
  position?: string;
}

export interface IReaction {
  id?: number;
  postId?: number;
  reactionName?: string;
  createdBy?: number;
  createdAt?: string;

  loading?: boolean;
  status?: 'pending' | 'success' | 'failed';
  localId?: string | number[]; // from uuid-v4
  parentCommentId?: string | number; // used when retry/cancel adding new comment
}

export interface IGetStreamAudienceUser {
  id?: number | string;
  collection?: string;
  foreign_id: string;
  data?: {
    avatar?: string;
    fullname?: string;
  };
  created_at: string;
  updated_at: string;
}

export interface IGetStreamAudienceGroup {
  id?: number | string;
  collection?: string;
  foreign_id: string;
  data?: {
    avatar?: string;
    icon?: string;
    name?: string;
  };
  created_at: string;
  updated_at: string;
}

export interface IGetStreamAudience {
  groups: IGetStreamAudienceGroup[];
  users: IGetStreamAudienceUser[];
}

export interface IGetStreamPost {
  id?: string;
  foreign_id?: string;
  actor: IGetStreamUser;
  audience: IGetStreamAudience;
  verb: string;
  type: string;
  data: IActivityData;
  object?: IGetStreamPost | string | any;
  origin?: string | null;
  target?: string;
  time?: string;
}

export interface IParamSearchMentionAudiences {
  key?: string;
  group_ids?: string;
  user_ids?: string;
  skip?: number;
  take?: number;
}

export interface IMentionUser {
  id: number;
  username: string;
  email?: string;
  fullname: string;
  avatar: string;
  bein_staff_role?: string;
  chat_user_id?: string;
}

export interface IParamGetReactionDetail {
  reactionType: ReactionType;
  postId?: string;
  commentId?: string;
  limit?: number;
  idLessThan?: string;
}

export interface IPostAudienceSheet {
  isShow: boolean;
  data: any[];
  fromStack?: 'newsfeed' | 'groups';
}

export interface IPayloadAddToAllPost {
  data: IPostActivity[] | IPostActivity;
  handleComment?: boolean;
}

export interface IPayloadReactToPost {
  id: string;
  reactionId: ReactionType;
  ownReaction: IOwnReaction;
  reactionCounts: IReactionCounts;
}

export interface IPayloadReactToComment {
  id: string;
  comment: IReaction;
  postId?: string;
  parentCommentId?: string;
  reactionId: ReactionType;
  ownReaction: IOwnReaction;
  reactionCounts: IReactionCounts;
}

export interface IParamPutReactionToPost {
  postId: string;
  data: string[];
}

export interface IParamPutReactionToComment {
  commentId: string;
  data: string[];
}

export interface IPayloadUpdateCommentsById {
  id: string;
  comments: IReaction[];
  isMerge: boolean;
}

export interface ICreatePostParams {
  draftPostId?: string;
  postId?: string;
  replaceWithDetail?: boolean;
  initAudience?: any;
  createFromGroupId?: number;
  initAutoSaveDraft?: boolean;
}

export interface IPayloadReplying {
  comment: IReaction;
  parentComment?: IReaction;
}

export interface IPayloadSetDraftPosts {
  posts?: IActivityData[];
  canLoadMore?: boolean;
  loading?: boolean;
  refreshing?: boolean;
}

export interface IParamGetDraftPosts {
  order?: 'ASC';
  offset?: number;
  limit?: number;
  idGTE?: number;
  idLTE?: number;
  idGT?: number;
  idLT?: number;
}

export interface IPayloadGetDraftPosts {
  isRefresh?: boolean;
  offset?: number;
}

export interface IPayloadPublishDraftPost {
  draftPostId: string;
  replaceWithDetail?: boolean;
  onSuccess?: () => void;
  onError?: () => void;
  refreshDraftPosts?: boolean;
  createFromGroupId?: number;
}

export interface IPayloadPutEditDraftPost {
  id: string;
  data: IPostCreatePost;
  replaceWithDetail?: boolean;
  publishNow: boolean;
  callback?: any;
  createFromGroupId?: number;
}

export interface IPayloadPutEditAutoSave {
  id: string;
  data: IPostCreatePost;
}

export interface IParamGetPostAudiences {
  key?: string;
  group_ids: string;
}

export interface IPayloadUpdateReaction {
  userId: string;
  data: ISocketReaction;
}

export interface ISocketReaction {
  actor: any;
  reaction: any;
  post: {
    post_id?: string;
    reaction_counts?: IReactionCounts;
    reactions_order?: string[];
  };
  comment: {
    comment_id?: string;
    reaction_counts?: IReactionCounts;
    reactions_order?: string[];
  };
}

export interface ICreatePostCurrentSettings {
  important: IActivityImportant;
}

export interface IPayloadDeleteComment {
  commentId: string;
  parentCommentId?: string;
  postId: string;
}
