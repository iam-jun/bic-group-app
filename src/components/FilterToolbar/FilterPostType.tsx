import { ExtendedTheme, useTheme } from '@react-navigation/native';
import React, { FC } from 'react';
import { TouchableOpacity, StyleSheet, View } from 'react-native';
import { useBaseHook } from '~/hooks';
import { spacing } from '~/theme';
import Text from '~/baseComponents/Text';
import { postTypeFilter } from './constants';
import { PostType } from '~/interfaces/IPost';
import { Button } from '~/baseComponents';
import Icon from '~/baseComponents/Icon';
import useModalStore from '~/store/modal';

type FilterPostTypeProps = {
  selectedPostType: PostType;
  onSelect: (postType: PostType) => void;
};

const FilterPostType: FC<FilterPostTypeProps> = ({
  selectedPostType,
  onSelect,
}) => {
  const { t } = useBaseHook();
  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = createStyles();
  const modalActions = useModalStore((state) => state.actions);

  const onSelectItemFilter = (item: any) => {
    onSelect?.(item.key);
    modalActions.hideModal();
  };

  const renderFilter = () => (
    <>
      {postTypeFilter.map((item) => (
        <Button testID={`btn_post_type_${item.key}`} key={item.key} onPress={() => onSelectItemFilter(item)}>
          <View style={styles.rowItemFilter}>
            <Text.BodyM useI18n color={colors.neutral40}>{item.text}</Text.BodyM>
            {selectedPostType === item.key && (
              <Icon icon="CircleCheckSolid" tintColor={colors.blue50} />
            )}
          </View>
        </Button>
      ))}
    </>
  );

  return (
    <TouchableOpacity testID="filter_post_type" activeOpacity={1} style={styles.container}>
      <Text.H4 style={styles.textHeader}>
        {t('home:newsfeed_search:filter_post_type')}
      </Text.H4>
      {renderFilter()}
    </TouchableOpacity>
  );
};

const createStyles = () => StyleSheet.create({
  container: {
    paddingBottom: spacing.padding.extraLarge,
  },
  textHeader: {
    marginTop: spacing.margin.tiny,
    marginBottom: spacing.margin.extraLarge,
    marginHorizontal: spacing.margin.large,
  },
  rowItemFilter: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: spacing.margin.extraLarge,
    paddingHorizontal: spacing.padding.large,
  },
});

export default FilterPostType;
