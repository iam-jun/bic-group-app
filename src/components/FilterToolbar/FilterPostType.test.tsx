import React from 'react';
import { PostType } from '~/interfaces/IPost';
import {
  act, fireEvent, renderHook, renderWithRedux,
} from '~/test/testUtils';
import FilterPostType from './FilterPostType';
import useFilterToolbarStore from './store';

describe('FilterPostType component', () => {
  it('should render correctly', () => {
    const { result } = renderHook(() => useFilterToolbarStore((state) => state));
    const { postType } = result.current;
    const onSelect = jest.fn();

    const rendered = renderWithRedux(<FilterPostType selectedPostType={postType} onSelect={onSelect} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('should render with init data correctly', () => {
    const { result } = renderHook(() => useFilterToolbarStore((state) => state));

    act(() => {
      result.current.actions.setFilterPostType(PostType.ARTICLE);
    });

    const { postType } = result.current;
    const onSelect = jest.fn();

    const rendered = renderWithRedux(<FilterPostType selectedPostType={postType} onSelect={onSelect} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('should pressable', () => {
    const { result } = renderHook(() => useFilterToolbarStore((state) => state));
    const { postType } = result.current;
    const onSelect = jest.fn();

    const rendered = renderWithRedux(<FilterPostType selectedPostType={postType} onSelect={onSelect} />);

    const btnPost = rendered.getByTestId(`btn_post_type_${PostType.POST}`);
    const btnArticle = rendered.getByTestId(`btn_post_type_${PostType.ARTICLE}`);
    const btnSeries = rendered.getByTestId(`btn_post_type_${PostType.SERIES}`);

    fireEvent.press(btnPost);
    expect(onSelect).toBeCalled();

    fireEvent.press(btnArticle);
    expect(onSelect).toBeCalled();

    fireEvent.press(btnSeries);
    expect(onSelect).toBeCalled();
  });
});
