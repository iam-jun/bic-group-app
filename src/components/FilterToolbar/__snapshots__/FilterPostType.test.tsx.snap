// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterPostType component should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingBottom": 24,
    }
  }
  testID="filter_post_type"
>
  <Text
    style={
      Array [
        Object {
          "color": "#181C32",
          "fontFamily": "BeVietnamPro-SemiBold",
          "fontSize": 17,
        },
        Object {},
        Object {
          "marginBottom": 24,
          "marginHorizontal": 16,
          "marginTop": 4,
        },
      ]
    }
  >
    Content type
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="btn_post_type_POST"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 24,
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#212745",
              "fontFamily": "BeVietnamPro-Light",
              "fontSize": 14,
            },
            Object {
              "color": "#353E6E",
            },
            undefined,
          ]
        }
      >
        Post
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="btn_post_type_ARTICLE"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 24,
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#212745",
              "fontFamily": "BeVietnamPro-Light",
              "fontSize": 14,
            },
            Object {
              "color": "#353E6E",
            },
            undefined,
          ]
        }
      >
        Article
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="btn_post_type_SERIES"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 24,
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#212745",
              "fontFamily": "BeVietnamPro-Light",
              "fontSize": 14,
            },
            Object {
              "color": "#353E6E",
            },
            undefined,
          ]
        }
      >
        Series
      </Text>
    </View>
  </View>
</View>
`;

exports[`FilterPostType component should render with init data correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingBottom": 24,
    }
  }
  testID="filter_post_type"
>
  <Text
    style={
      Array [
        Object {
          "color": "#181C32",
          "fontFamily": "BeVietnamPro-SemiBold",
          "fontSize": 17,
        },
        Object {},
        Object {
          "marginBottom": 24,
          "marginHorizontal": 16,
          "marginTop": 4,
        },
      ]
    }
  >
    Content type
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="btn_post_type_POST"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 24,
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#212745",
              "fontFamily": "BeVietnamPro-Light",
              "fontSize": 14,
            },
            Object {
              "color": "#353E6E",
            },
            undefined,
          ]
        }
      >
        Post
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="btn_post_type_ARTICLE"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 24,
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#212745",
              "fontFamily": "BeVietnamPro-Light",
              "fontSize": 14,
            },
            Object {
              "color": "#353E6E",
            },
            undefined,
          ]
        }
      >
        Article
      </Text>
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              undefined,
              Object {
                "backgroundColor": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <View />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="btn_post_type_SERIES"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 24,
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#212745",
              "fontFamily": "BeVietnamPro-Light",
              "fontSize": 14,
            },
            Object {
              "color": "#353E6E",
            },
            undefined,
          ]
        }
      >
        Series
      </Text>
    </View>
  </View>
</View>
`;
